app/ide-desktop/lib/dashboard/src/providers/AuthProvider.tsx

```diff
@@ -249,9 +249,29 @@ export default function AuthProvider(props: AuthProviderProps) {
       platform: detect.platform(),
       architecture: detect.architecture(),
     })
+
     return gtagHooks.gtagOpenCloseCallback(gtagEventRef, 'open_app', 'close_app')
   }, [])
 
+  const remoteBackend = React.useMemo(() => {
+    if (session) {
```

Comment:
Looks like a cycle dependency(we have to create a remoteBackend in two different places)

Reply:
It is specifically avoiding a cycle dependency on two providers. I've added a commend explaining the necessity of this duplication for now. It can potentially be removed by refactoring the auth provider to separate some responsibilities away from it, but this is very out of scope for this PR.